--- src/vs/platform/windows/electron-main/windowImpl.ts	2022-12-02 08:39:41.678546844 +0100
+++ src/vs/platform/windows/electron-main/windowImpl.ts	2022-12-02 08:42:03.560297136 +0100
@@ -238,7 +238,7 @@
 				backgroundColor: this.themeMainService.getBackgroundColor(),
 				minWidth: WindowMinimumSize.WIDTH,
 				minHeight: WindowMinimumSize.HEIGHT,
-				show: !isFullscreenOrMaximized, // reduce flicker by showing later
+				show: false,
 				title: this.productService.nameLong,
 				webPreferences: {
 					preload: FileAccess.asFileUri('vs/base/parts/sandbox/electron-browser/preload.js').fsPath,
@@ -418,7 +418,7 @@
 
 				// to reduce flicker from the default window size
 				// to maximize or fullscreen, we only show after
-				this._win.show();
+				// this._win.show();
 				mark('code/didMaximizeCodeWindow');
 			}
 
@@ -982,7 +982,7 @@
 		if (!this.environmentMainService.isBuilt && !this.environmentMainService.extensionTestsLocationURI) {
 			this._register(new RunOnceScheduler(() => {
 				if (this._win && !this._win.isVisible() && !this._win.isMinimized()) {
-					this._win.show();
+					// this._win.show();
 					this.focus({ force: true });
 					this._win.webContents.openDevTools();
 				}
